src/InterpreterEngine.cpp

```diff
@@ -1477,4 +1219,138 @@ RamDomain InterpreterEngine::execute(const InterpreterNode* node, InterpreterCon
+            shouldRunNested = true;
+            break;
+        case AggregateOp::USUM:
+            res = ramBitCast(static_cast<RamUnsigned>(0));
+            shouldRunNested = true;
+            break;
+        case AggregateOp::FSUM:
+            res = ramBitCast(static_cast<RamFloat>(0));
+            shouldRunNested = true;
+            break;
+
+        case AggregateOp::MEAN:
+            res = 0;
+            accumulateMean = {0, 0};
+            break;
+
+        case AggregateOp::COUNT:
+            res = 0;
+            shouldRunNested = true;
+            break;
+    }
+
+    for (auto ip : stream) {
+        const RamDomain* data = &ip[0];
+        ctxt[aggregate.getTupleId()] = data;
+
+        if (!execute(&filter, ctxt)) {
+            continue;
+        }
+
+        shouldRunNested = true;
+
+        // count is a special case.
+        if (aggregate.getFunction() == AggregateOp::COUNT) {
+            ++res;
+            continue;
+        }
+
+        // eval target expression
+        RamDomain val = execute(&expression, ctxt);
+
+        switch (aggregate.getFunction()) {
+            case AggregateOp::MIN:
+                res = std::min(res, val);
+                break;
+            case AggregateOp::FMIN:
+                res = ramBitCast(std::min(ramBitCast<RamFloat>(res), ramBitCast<RamFloat>(val)));
+                break;
+            case AggregateOp::UMIN:
+                res = ramBitCast(std::min(ramBitCast<RamUnsigned>(res), ramBitCast<RamUnsigned>(val)));
+                break;
+
+            case AggregateOp::MAX:
+                res = std::max(res, val);
+                break;
+            case AggregateOp::FMAX:
+                res = ramBitCast(std::max(ramBitCast<RamFloat>(res), ramBitCast<RamFloat>(val)));
+                break;
+            case AggregateOp::UMAX:
+                res = ramBitCast(std::max(ramBitCast<RamUnsigned>(res), ramBitCast<RamUnsigned>(val)));
+                break;
+
+            case AggregateOp::SUM:
+                res += val;
+                break;
+            case AggregateOp::FSUM:
+                res = ramBitCast(ramBitCast<RamFloat>(res) + ramBitCast<RamFloat>(val));
+                break;
+            case AggregateOp::USUM:
+                res = ramBitCast(ramBitCast<RamUnsigned>(res) + ramBitCast<RamUnsigned>(val));
+                break;
+
+            case AggregateOp::MEAN:
+                accumulateMean.first += ramBitCast<RamFloat>(val);
+                accumulateMean.second++;
+                break;
+
+            case AggregateOp::COUNT:
+                assert(false && "This should never be executed");
+                break;
+        }
+    }
+
+    if (aggregate.getFunction() == AggregateOp::MEAN && accumulateMean.second != 0) {
+        res = ramBitCast(accumulateMean.first / accumulateMean.second);
+    }
+
+    // write result to environment
+    RamDomain tuple[1];
+    tuple[0] = res;
+    ctxt[aggregate.getTupleId()] = tuple;
```

Comment:
Nitpick: `ctxt[aggregate.getTupleId()] = &res;` should do, no need to introduce another local.